xarray/core/indexing.py

```diff
@@ -217,23 +217,23 @@ def expanded_indexer(key, ndim):
     if not isinstance(key, tuple):
         # numpy treats non-tuple keys equivalent to tuples of length 1
         key = (key,)
-    new_key = []
+    new_key = ()
```

Comment:
So the bad news is that @andersy005 is refactoring a lot of this file (`indexing.py`) in the `backend-indexing` branch. Would you mind pushing these changes in a PR to that branch? I fear the merge conflicts will be quite complex otherwise.

Reply:
That ok, I think I have most of the changes contained in a single commit: https://github.com/pydata/xarray/pull/9002/commits/5328f6c9b4c90d02bf450082d16cd9c43f91d57d